tracer/test/Datadog.Trace.Tools.dd_dotnet.ArtifactTests/CreatedumpTests.cs

```diff
@@ -0,0 +1,343 @@
+
+        await helper.Task;
+
+        using var assertionScope = new AssertionScope();
+        assertionScope.AddReportable("stdout", helper.StandardOutput);
+        assertionScope.AddReportable("stderr", helper.ErrorOutput);
+
+        helper.StandardOutput.Should().NotContain(CrashReportExpectedOutput);
+
+        if (enableCrashDumps)
+        {
+            helper.StandardOutput.Should().Contain(CreatedumpExpectedOutput);
+        }
+        else
+        {
+            helper.StandardOutput.Should().NotContain(CreatedumpExpectedOutput);
+        }
+
+        File.Exists(reportFile.Path).Should().BeFalse();
+    }
+
+    [SkippableTheory]
+    [InlineData(true, true)]
+    [InlineData(false, true)]
+    [InlineData(false, false)]
+    public async Task DisableTelemetry(bool telemetryEnabled, bool crashdumpEnabled)
+    {
+        SkipOn.Platform(SkipOn.PlatformValue.MacOs);
+
+        using var reportFile = new TemporaryFile();
+
+        (string, string)[] args = [LdPreloadConfig, ..CrashReportConfig(reportFile)];
+
+        if (crashdumpEnabled)
+        {
+            args = [..args, ..CreatedumpConfig];
+        }
+
+        args = [..args, ("DD_INSTRUMENTATION_TELEMETRY_ENABLED", telemetryEnabled ? "1" : "0")];
+
+        using var helper = await StartConsoleWithArgs("crash-datadog", args);
+
+        await helper.Task;
+
+        using var assertionScope = new AssertionScope();
+        assertionScope.AddReportable("stdout", helper.StandardOutput);
+        assertionScope.AddReportable("stderr", helper.ErrorOutput);
+
+        if (crashdumpEnabled)
+        {
+            helper.StandardOutput.Should().Contain(CreatedumpExpectedOutput);
+        }
+        else
+        {
+            helper.StandardOutput.Should().NotContain(CreatedumpExpectedOutput);
+        }
+
+        if (telemetryEnabled)
+        {
+            helper.StandardOutput.Should().Contain(CrashReportExpectedOutput);
+            File.Exists(reportFile.Path).Should().BeTrue();
+        }
+        else
+        {
+            helper.StandardOutput.Should().NotContain(CrashReportExpectedOutput);
+            File.Exists(reportFile.Path).Should().BeFalse();
+        }
+    }
+
+    [SkippableFact]
+    public async Task WriteCrashReport()
+    {
+        SkipOn.Platform(SkipOn.PlatformValue.MacOs);
+
+        using var reportFile = new TemporaryFile();
+
+        using var helper = await StartConsoleWithArgs(
+                               "crash-datadog",
+                               [LdPreloadConfig, ..CrashReportConfig(reportFile)]);
+
+        await helper.Task;
+
+        using var assertionScope = new AssertionScope();
+        assertionScope.AddReportable("stdout", helper.StandardOutput);
+        assertionScope.AddReportable("stderr", helper.ErrorOutput);
+
+        helper.StandardOutput.Should().Contain(CrashReportExpectedOutput);
+
+        File.Exists(reportFile.Path).Should().BeTrue();
+
+        var report = JObject.Parse(reportFile.GetContent());
```

Comment:
I wonder if it's worth snapshot testing this? Probably not due to too much variation in runtimes, but just a thought 🤷‍♂️

Reply:
Almost everything will be different from one run to the other, so yuck